azure/services/agentpools/spec.go

```diff
@@ -331,97 +254,37 @@ func (s *AgentPoolSpec) Parameters(ctx context.Context, existing interface{}) (p
-			EnableUltraSSD:       s.EnableUltraSSD,
-			KubeletConfig:        kubeletConfig,
-			KubeletDiskType:      azure.AliasOrNil[armcontainerservice.KubeletDiskType]((*string)(s.KubeletDiskType)),
-			MaxCount:             s.MaxCount,
-			MaxPods:              s.MaxPods,
-			MinCount:             s.MinCount,
-			Mode:                 ptr.To(armcontainerservice.AgentPoolMode(s.Mode)),
-			NodeLabels:           nodeLabels,
-			NodeTaints:           nodeTaints,
-			OrchestratorVersion:  s.Version,
-			OSDiskSizeGB:         &s.OSDiskSizeGB,
-			OSDiskType:           azure.AliasOrNil[armcontainerservice.OSDiskType](s.OsDiskType),
-			OSType:               azure.AliasOrNil[armcontainerservice.OSType](s.OSType),
-			ScaleSetPriority:     azure.AliasOrNil[armcontainerservice.ScaleSetPriority](s.ScaleSetPriority),
-			ScaleDownMode:        azure.AliasOrNil[armcontainerservice.ScaleDownMode](s.ScaleDownMode),
-			SpotMaxPrice:         spotMaxPrice,
-			Type:                 ptr.To(armcontainerservice.AgentPoolTypeVirtualMachineScaleSets),
-			VMSize:               sku,
-			VnetSubnetID:         vnetSubnetID,
-			EnableNodePublicIP:   s.EnableNodePublicIP,
-			NodePublicIPPrefixID: s.NodePublicIPPrefixID,
-			Tags:                 tags,
-			EnableFIPS:           s.EnableFIPS,
-			LinuxOSConfig:        linuxOSConfig,
-		},
+	// When autoscaling is set, the count of the nodes differ based on the autoscaler and should not depend on the
+	// count present in MachinePool or AzureManagedMachinePool, hence we should not make an update API call based
+	// on difference in count.
+	if s.EnableAutoScaling && agentPool.Status.Count != nil {
+		spec.Count = agentPool.Status.Count
 	}
 
 	return agentPool, nil
 }
 
-// mergeSystemNodeLabels appends any kubernetes.azure.com-prefixed labels from the AKS label set
-// into the local capz label set.
-func mergeSystemNodeLabels(capz, aks map[string]*string) map[string]*string {
-	ret := capz
-	if ret == nil {
-		ret = make(map[string]*string)
-	}
-	// Look for labels returned from the AKS node pool API that begin with kubernetes.azure.com
-	for aksNodeLabelKey := range aks {
-		if azureutil.IsAzureSystemNodeLabelKey(aksNodeLabelKey) {
-			ret[aksNodeLabelKey] = aks[aksNodeLabelKey]
-		}
-	}
-	// Preserve nil-ness of capz
-	if capz == nil && len(ret) == 0 {
-		ret = nil
-	}
-	return ret
-}
-
-// mergeSystemNodeTaints appends any kubernetes.azure.com-prefixed taints from the AKS taint set
-// into the local capz taint set.
-func mergeSystemNodeTaints(capz, aks []*string) []*string {
-	ret := capz
-	if ret == nil {
-		ret = make([]*string, 0)
-	}
-	// Look for taints returned from the AKS node pool API that begin with kubernetes.azure.com
-	for _, aksNodeTaint := range aks {
-		if azureutil.IsAzureSystemNodeLabelKey(*aksNodeTaint) {
-			ret = append(ret, aksNodeTaint)
-		}
-	}
-	// Preserve nil-ness of capz
-	if capz == nil && len(ret) == 0 {
-		ret = nil
-	}
-	return ret
+// WasManaged implements azure.ASOResourceSpecGetter.
```

Comment:
Should we clarify in the comment that it always returns true because we don't support byo agentpools?